Allow matching event to pipeline by modified files for GitHub pull request

savitaashture opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes
* Added changes to trigger pipelinerun for matched files using custom CEL functions for **GitHub Pull Request**

Gif:
https://drive.google.com/drive/folders/1-JYHVYtoh1kyqa05ekAWltFmuDa2g6_K?usp=sharing

/cc @chmouel @sm43 

<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

- [ ] ♽  Run `make test lint` before submitting a PR (ie: with [pre-commit](https://pipelinesascode.com/dev/tools), no need to waste CPU cycle on CI
- [ ] 📖 If you are adding a user facing feature or make a change of the behavior, please verify that you have documented it
- [ ] 🧪 100% coverage is not a target but most of the time we would rather have a unit test if you make a code change.
- [ ] 🎁 If that's something that is possible to do please ensure to check if we can add a e2e test.
- [ ] 🔎 If there is a flakiness in the CI tests then don't *necessary* ignore it, better get the flakyness fixed before merging or if that's not possible there is a good reason to bypass it. (token rate limitation may be a good reason to skip).

savitaashture committed:
Allow matching event to pipeline by modified files for GitHub pull request

chmouel commented:
/retest

chmouel commented:
/retest

sm43 commented on pkg/matcher/cel.go:
```diff
@@ -1,22 +1,30 @@
 )
 
-func celEvaluate(expr string, event *info.Event) (ref.Val, error) {
+func celEvaluate(expr string, event *info.Event, vcx provider.Interface, ctx context.Context) (ref.Val, error) {
```
```suggestion
func celEvaluate(ctx context.Context, expr string, event *info.Event, vcx provider.Interface) (ref.Val, error) {
```

sm43 commented on pkg/matcher/cel.go:
```diff
@@ -46,3 +54,43 @@ func celEvaluate(expr string, event *info.Event) (ref.Val, error) {
 	return out, nil
 }
+
+type celPac struct {
```
may be some unit tests?

sm43 replied:
we can use `TestProviderImp` for mocking provider

chmouel committed:
Update pkg/matcher/cel.go

Co-authored-by: Shivam Mukhade <smukhade@redhat.com>

chmouel committed:
Fix linting errors

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

chmouel committed:
Revert cel-go to 0.11.x

since it breaks cel functions

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

chmouel commented:
Updated PR

still need unit/e2e

chmouel committed:
Improve pathChanged CEL function

* make it work with globbing
* add unit tests
* add docs

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

chmouel committed:
add context to http.NewRequest

fix golangci-lint errors for latest (which is useful error too)

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

chmouel commented:
modified to use globs (ie: `".tekton/*yaml".hasChanged`) add docs and test

chmouel committed:
fix bad comparaison

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

chmouel committed:
Add support for gitlab Getfiles

not tested because we need to change how we do test for annotation
matcher to not be tight to github.

Signed-off-by: Chmouel Boudjnah <chmouel@redhat.com>

chmouel commented:
the e2e errors is ratelimit on token

savitaashture commented:
/lgtm

Thank you :bowing_woman:

chmouel commented:
thanks for the review, will merge when the cluster is back, (13h00 CEST is reinstalling) it if you have sometime to review #769 too that would be great 🙇🏻
